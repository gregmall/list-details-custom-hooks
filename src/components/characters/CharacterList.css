.display{
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  width: 100%;
  margin: 10px;
  font-family:'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  
  
  
}
.item{
  padding: 0 10px;
  list-style: none;
  
}
