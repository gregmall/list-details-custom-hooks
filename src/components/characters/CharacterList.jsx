import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom'
import { useCharacters } from '../../hooks/characters';
import Character from './Character'

const CharacterList = ({ page }) => {
  const { loading, characters } = useCharacters(page);
  if(loading) return <span><img src="http://cdn.lowgif.com/full/ff8280aafe27319d-ajax-loading-gif-transparent-background-2-gif-images.gif"/></span>


  const characterElements = characters.map(character => (
  
      <li key={character.id}>
        <Link to = {`/detail/${character.id}`}>
          <Character  {...character } />
        </Link>
      </li>
  
    ));

  return (
    <ul data-testid="characters">
      {characterElements}
    </ul> 
  );
};

CharacterList.propTypes = {
 page: PropTypes.number
};


export default CharacterList;

