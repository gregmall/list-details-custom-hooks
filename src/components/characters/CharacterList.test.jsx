  
import React from 'react';
import { render } from '@testing-library/react';
import CharacterList from './CharacterList';

describe('List Component', () => {
  const characters = [{ name: 'rick', image: 'www.url.com' }, { name: 'morty', image: 'www.page.com' }];

  it('renders List component', () => {
    const { asFragment } = render(<CharacterList characters={characters} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
