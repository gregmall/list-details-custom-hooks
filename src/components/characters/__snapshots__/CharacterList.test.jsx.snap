// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component renders List component 1`] = `
<DocumentFragment>
  <span>
    <img
      src="http://cdn.lowgif.com/full/ff8280aafe27319d-ajax-loading-gif-transparent-background-2-gif-images.gif"
    />
  </span>
</DocumentFragment>
`;
